### ISSUE TITLE
Hardware keyboard focus trap on iOS when tabbing from TextInput to Button

### ISSUE THREAD

**Issue by kriti18singh** [facebook/react-native#31853]
## Description
I have 2 TextInput components and a button in my react native app(layout looks like screenshot attached). Trying to tab using a hardware keyboard attached to my iOS app causes keyboard focus to get trapped between the TextInput components. Tab or Shift+Tab only moves the focus from TextInput 1 to TextInput 2 back and forth and never reaches the Button. Replacing the button with TouchableOpacity or a View (with focusable set to true) does not work either. 

The docs don't have any explanation on how to get the hardware keyboard working. This is important for accessibility requirements. Is this the expected behaviour since I can not find an open bug on this?
Any help is greatly appreciated.

## React Native version:
System:
    OS: macOS 11.4
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 54.05 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 12.18.3 - /usr/local/opt/node@12/bin/node
    Yarn: 1.22.5 - /usr/local/bin/yarn
    npm: 6.14.6 - /usr/local/opt/node@12/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.10.1 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.5, DriverKit 20.4, macOS 11.3, tvOS 14.5, watchOS 7.4
    Android SDK:
      API Levels: 26, 28, 29, 30
      Build Tools: 28.0.3, 29.0.2, 29.0.3, 30.0.2
      System Images: android-25 | Google APIs ARM 64 v8a, android-27 | Google APIs Intel x86 Atom, android-27 | Google Play Intel x86 Atom, android-28 | Intel x86 Atom, android-28 | Intel x86 Atom_64, android-28 | Google APIs Intel x86 Atom, android-28 | Google APIs Intel x86 Atom_64, android-28 | Google Play Intel x86 Atom, android-28 | Google Play Intel x86 Atom_64, android-29 | Intel x86 Atom_64, android-29 | Google APIs Intel x86 Atom, android-29 | Google APIs Intel x86 Atom_64, android-29 | Google Play Intel x86 Atom, android-30 | Google APIs Intel x86 Atom, android-30 | Google APIs Intel x86 Atom_64, android-30 | Google Play Intel x86 Atom, android-30 | Google Play Intel x86 Atom_64
      Android NDK: Not Found
  IDEs:
    Android Studio: 4.0 AI-193.6911.18.40.6626763
    Xcode: 12.5.1/12E507 - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_265 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.1 => 17.0.1 
    react-native: 0.64.2 => 0.64.2 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

## Steps To Reproduce
Provide a detailed list of steps that reproduce the issue.

1. Attach a hardware keyboard to your device (usb or bluetooth)
2. Use tab key to navigate the app.
3. Observe that the keyboard focus never moves to the Button.

## Expected Results
Tabbing should allow the user to navigate from the TextInput to Button

## Snack, code example, screenshot, or link to a repository:
Here is what the screen looks like 
<img width="299" alt="Screenshot 2021-07-13 at 12 29 03 PM" src="https://user-images.githubusercontent.com/18608953/125412576-ad9d2780-e3dc-11eb-86e4-11118d8611f5.png">

I have reproduced the issue in the simplest test app where the code is like below : 
```
<SafeAreaView style={backgroundStyle}>
      <TextInput placeholder={'Enter your name'} />
      <TextInput
        style={{paddingTop: 20}}
        placeholder={'Enter your address'}
      />
      <Button title="Submit" />
</SafeAreaView>
```

**Comment by rectified95** [comment]
@heyimchris Does this work on MacOS, and if so, could we port it to iOS?

**Comment by kriti18singh** [comment]
@rectified95 @HeyImChris Is there any update on this?

**Comment by HeyImChris** [comment]
I can tab to Buttons when the `onPress` prop is set. If I comment out that then the button is no longer navigable. Seems like a bug that buttons aren't always in Apple's view hierarchy but if you're blocked on this, have you tried adding an onPress like below to make the button navigable?

```
<Button
  // onPress={() => {
  //   if (this.inputRef != null) {
  //     this.inputRef.clear();
  //   }
  // }}
  title="Submit"
/>
```

**Comment by kriti18singh** [comment]
@HeyImChris This does not work for me. Which version are you on?
